modules/Channel.md

```diff
@@ -0,0 +1,112 @@
+## Channel Module
+
+A channel is a FIFO data structure used to communicate between different Chapel
+tasks. You can learn more about how go-channels work by following this
+[link](https://gobyexample.com/channels)
+Since the operations on the channel are blocking by default, it enables to send
+values from one task to another in a synchronized manner.
+Here is a simple example in Go.
+
+```
+func main() {
+	// make a channel for string values
+	messages := make(chan string)
+
+	// new task to send a value to the channel
+	// it will wait until other task receives
+	go func() { messages <- "ping" }()
+
+	// receive the sent value
+	msg := <-messages
+	fmt.Println(msg)
+}
+```
+
+The module will have a generic `Channel` class (or record) to store the channel
+buffer and support operations on them.
+
+* The main functions that the module should have
+
+	* `init(type eltType, size : int)` : Specify the buffer data type and size
+	for creating a Channel.
+
+	* `send(value : eltType)` : This function waits for a space in the buffer
+	and	sends the `value` to the channel.
+
+	* `recv(out value : eltType) : bool` : This function waits until there is
+	something in the buffer and stores the received entity in `value`. It also
+	returns the status i.e., the operation is executed successfully or not.
+
+	* `close()` : This function is used to close the channel such that no more
+	values will be sent to it.
+
+* `iter these()` : An iterator to receive the values sent to the channel until
+it is closed.
+
+* `select` statements : This statement lets you wait for multiple channel
+operations and executes if anyone of them is ready for communication. `select`
+statements in Go are specifically designed to work with channels.
+Some key points are:
+
+	* `select` statements are commonly used in Go code when working with channels.
+	* The operations are atomic, i.e., checking a channel and performing `recv`
+		or `send` on it is considered as one operation.
+	* The cases are arranged in a random order to prevent starvation.
```

Comment:
"arranged" in a random order (suggesting to me that their order in the source code is random)? or "evaluated"/"selected from" in a random order (and: just among those that can execute?  or is the randomness applied to all of them to see whether they can execute or not?)

grammar: "anyone" -> "any one"

Reply:
Yeah, I think the "evaluated" would be a more appropriate word here. In the implementation, the cases are first checked in a random order if they are ready or not (if ready, then that case is executed). It is done because if we run a particular `select` block multiple times, it should not happen that at each iteration, we are performing operations on only one channel, and others are kept waiting.